spacy_llm/tests/tasks/test_entity_linker.py

```diff
@@ -0,0 +1,622 @@
+    cfg_string = request.getfixturevalue(cfg_string)
+    config = Config().from_str(
+        cfg_string,
+        overrides={
+            "paths.el_nlp": str(tmp_path),
+            "paths.el_desc": str(tmp_path / "desc.csv"),
+        },
+    )
+    build_el_pipeline(nlp_path=tmp_path, desc_path=tmp_path / "desc.csv")
+    nlp = spacy.util.load_model_from_config(config, auto_fill=True)
+    assert nlp.pipe_names == ["llm"]
+
+    # also test nlp config from a dict in add_pipe
+    component_cfg = dict(config["components"]["llm"])
+    component_cfg.pop("factory")
+
+    nlp2 = spacy.blank("en")
+    nlp2.add_pipe("llm", config=component_cfg)
+    assert nlp2.pipe_names == ["llm"]
+
+
+@pytest.mark.external
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+@pytest.mark.parametrize(
+    "cfg_string",
+    [
+        "zeroshot_cfg_string",
+        "fewshot_cfg_string",
+        "ext_template_cfg_string",
+    ],
+)
+def test_entity_linker_predict(cfg_string, request, tmp_path):
+    """Use OpenAI to get EL results."""
+    cfg = request.getfixturevalue(cfg_string)
+    orig_config = Config().from_str(
+        cfg,
+        overrides={
+            "paths.el_nlp": str(tmp_path),
+            "paths.el_desc": str(tmp_path / "desc.csv"),
+        },
+    )
+    build_el_pipeline(nlp_path=tmp_path, desc_path=tmp_path / "desc.csv")
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
+    nlp.initialize(lambda: [])
+
+    text = "Alice goes to Boston to see the Boston Celtics game."
+    doc = nlp.make_doc(text)
+    doc.ents = [
+        Span(doc=doc, start=3, end=4, label="LOC"),  # Q100
+        Span(doc=doc, start=7, end=9, label="ORG"),  # Q131371
+    ]
+    doc = nlp(doc)
+    if cfg_string != "ext_template_cfg_string":
+        assert len(doc.ents) == 2
+        assert doc.ents[0].kb_id_ == "Q100"
+        assert doc.ents[1].kb_id_ == "Q131371"
+
+
+@pytest.mark.external
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+def test_entity_linker_predict_no_candidates(request, tmp_path):
+    """Test behavior if no candidates are available for certain mention."""
+    cfg = request.getfixturevalue("zeroshot_cfg_string")
+    orig_config = Config().from_str(
+        cfg,
+        overrides={
+            "paths.el_nlp": str(tmp_path),
+            "paths.el_desc": str(tmp_path / "desc.csv"),
+        },
+    )
+    build_el_pipeline(nlp_path=tmp_path, desc_path=tmp_path / "desc.csv")
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
+    nlp.initialize(lambda: [])
+
+    text = "Alice goes to Foo to see the Boston Celtics game."
+    doc = nlp.make_doc(text)
+    doc.ents = [
+        Span(doc=doc, start=3, end=4, label="LOC"),  # NIL
+        Span(doc=doc, start=7, end=9, label="ORG"),  # Q131371
+    ]
+    doc = nlp(doc)
+
+    assert len(doc.ents) == 2
+    assert doc.ents[0].kb_id_ == "NIL"
```

Comment:
```suggestion
    assert doc.ents[0].kb_id_ == EntityLinker.NIL
```
Also, add the import. As the component specifically sets this to `EntityLinker.NIL`, we should test for it like that, too, right?